debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/actions/CDTDebugPropertyTester.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Advantest Europe GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * 				Raghunandana Murthappa
+ *******************************************************************************/
+package org.eclipse.cdt.debug.internal.ui.actions;
+
+import org.eclipse.cdt.debug.core.model.ISteppingModeTarget;
+import org.eclipse.core.expressions.PropertyTester;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.debug.core.model.IDebugElement;
+import org.eclipse.debug.core.model.IDebugTarget;
+import org.eclipse.debug.ui.IDebugUIConstants;
+import org.eclipse.debug.ui.IDebugView;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.TreeSelection;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+
+/**
+ *Tests whether an active C/C++ application is debugging. And selection inside Debug View is present on it.
+ *
+ *@author Raghunandana Murthappa
+ */
+public class CDTDebugPropertyTester extends PropertyTester {
+
+	public static final String IS_CDT_DEBUGGING = "isCDTDebugging"; //$NON-NLS-1$
+
+	@Override
+	public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
+		if (IS_CDT_DEBUGGING.equals(property)) {
+			return isCppAppDebugging();
+		}
+		return false;
+	}
+
+	private boolean isCppAppDebugging() {
```

Comment:
I hve named it as CppApp because it is any C/C++ application under debug.
It can also be something like "IsLaunchDebugging".
Can you suggest any better naming here please ?

Reply:
I would rather call it something with "CDT" if you need some kind of special naming.
Cpp can be interpreted as a C++ only thing and not as C/C++.